### PR TITLE
Use @actions/attest and @sigstore/oci libraries

### PR DESCRIPTION
**Pull request by bdehamer**
Updates the action logic to use the `@actions/attest` and `@sigstore/oci` libraries instead of the local implementation.

Among other things, this brings-in the logic which generates the provenance statement from the GHA OIDC token instead of from the runtime environment.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -59,39 +63,46 @@ export async function run(): Promise<void> {
...
-  const provenance = generateProvenance(subject, process.env)
-
+  const predicate = await buildSLSAProvenancePredicate()
   core.startGroup(
     highlight(
       `Provenance attestation generated for ${subject.name} (sha256:${subject.digest.sha256})`
     )
   )
-  core.info(JSON.stringify(provenance, null, 2))
+  core.info(
```
Comment (malancas) [inline src/main.ts:76]: What does the info function do? Does it print content to the console during an Actions run?

Comment (bdehamer) [reply src/main.ts:76]: Yeah, the info function allows us to write to the job ouput:
image

File: src/main.ts [hunk 2]
```diff
@@ -59,39 +63,46 @@ export async function run(): Promise<void> {
...
 
-const attest = async (
+const createAttestation = async (
   subject: Subject,
   visibility: Visibility
 ): Promise<Attestation> => {
-  const provenance = generateProvenance(subject, process.env)
-
+  const predicate = await buildSLSAProvenancePredicate()
   core.startGroup(
```
[the remark this answers is not in the document]
Comment (bdehamer) [reply src/main.ts:71]: The start/endGroup functions render toggle'able regions in the output:
image

**Review by malancas — APPROVED** [review]
LGTM ~

**Pull request merged**